pkg/jx/cmd/install.go

```diff
@@ -923,321 +909,815 @@ func (options *InstallOptions) Run() error {
 	}
 
-	valueFiles := []string{cloudEnvironmentValuesLocation, secretsFileName, adminSecretsFileName, configFileName, cloudEnvironmentSecretsLocation}
-	valueFiles, err = helm.AppendMyValues(valueFiles)
+	err := options.configureTillerNamespace()
 	if err != nil {
-		return errors.Wrap(err, "failed to append the myvalues.yaml file")
+		return errors.Wrap(err, "configuring the tiller namespace")
 	}
 
-	options.currentNamespace = ns
-	if options.Flags.Prow {
-		// install Prow into the new env
-		options.OAUTHToken = options.GitRepositoryOptions.ApiToken
-		err = options.installProw()
+	isProw := false
+	if options.Flags.GitOpsMode {
+		isProw = options.Flags.Prow
+	} else {
+		options.SetDevNamespace(namespace)
+		isProw, err = options.isProw()
 		if err != nil {
-			return fmt.Errorf("failed to install Prow: %v", err)
+			return errors.Wrapf(err, "cannot work out if this is a prow based install in namespace %s", options.currentNamespace)
 		}
 	}
-	timeoutInt, err := strconv.Atoi(timeout)
-	if err != nil {
-		return errors.Wrap(err, "failed to convert the helm install timeout value")
+
+	if isProw {
+		enableJenkins := false
+		helmConfig.Jenkins.Enabled = &enableJenkins
 	}
-	options.Helm().SetCWD(makefileDir)
-	jxChart := JenkinsXPlatformChart
-	jxRelName := "jenkins-x"
 
-	log.Infof("Installing jx into namespace %s\n", util.ColorInfo(ns))
-	if !initOpts.Flags.NoTiller {
-		// Need to check the tiller pod is ready before proceeding
-		serviceAccountName := "tiller"
-		tillerNamespace := options.InitOptions.Flags.TillerNamespace
+	return nil
+}
+
+func (options *InstallOptions) getHelmValuesFiles(configStore configio.ConfigStore, providerEnvDir string) ([]string, []string, []string, error) {
+	helmConfig := &options.CreateEnvOptions.HelmValuesConfig
+	cloudEnvironmentValuesLocation := filepath.Join(providerEnvDir, CloudEnvValuesFile)
+	cloudEnvironmentSecretsLocation := filepath.Join(providerEnvDir, CloudEnvSecretsFile)
+
+	valuesFiles := []string{}
+	secretsFiles := []string{}
+	temporaryFiles := []string{}
+
+	adminSecrets, err := options.getAdminSecrets(providerEnvDir, cloudEnvironmentSecretsLocation)
+	if err != nil {
+		return valuesFiles, secretsFiles, temporaryFiles,
+			errors.Wrap(err, "creating the admin secrets")
+	}
+
+	gitSecrets, err := options.getGitSecrets()
+	if err != nil {
+		return valuesFiles, secretsFiles, temporaryFiles,
+			errors.Wrap(err, "failed to read the git secrets from configuration")
+	}
+
+	dir, err := util.ConfigDir()
+	if err != nil {
+		return valuesFiles, secretsFiles, temporaryFiles,
+			errors.Wrap(err, "failed to create a temporary config dir for Git credentials")
+	}
+	gitSecretsFileName := filepath.Join(dir, GitSecretsFile)
+	err = configStore.Write(gitSecretsFileName, []byte(gitSecrets))
+	if err != nil {
+		return valuesFiles, secretsFiles, temporaryFiles,
+			errors.Wrap(err, "failed to write the git secrets in the secrets file")
```

Comment:
For this and subsequent errors, it is probably worth putting the name of the file in the error